src/main/scala/util/AsyncQueue.scala

```diff
@@ -96,10 +98,10 @@ class AsyncQueueSink[T <: Data](gen: T, depth: Int, sync: Int) extends Module {
   // This is safe, because isolation gates brought mem low before the zeroed widx reached us
   io.deq.bits  := RegEnable(io.mem(index), valid)
 
-  val valid_reg = AsyncResetReg(valid.asUInt)(0)
+  val valid_reg = AsyncResetReg(valid.asUInt, "valid_reg")(0)
```

Comment:
If you want to add a Bool form to your AsyncResetReg stuff, I would use it.
However, I don't find the cast to be particularly evil.